File: matlab/group_test.m
% Скрипт тестирования группы сценариев обработка на наборе данных

% Очистка и инициализации класса хранения данных обработки
clear Storage
Storage = Storage();

% Определение каталога с данными и формата изображения
folder_address = 'demos/group_test';
image_format = '*.tif';

% Определение набора сценариев для расчета
processing_vec{1,1} = @processing_1;
processing_vec{2,1} = @processing_2;
processing_vec{3,1} = @processing_3;
processing_vec{4,1} = @processing_4;
processing_vec{5,1} = @processing_5;

% Сравнение векторных полей из flow файлов с результатами сценариев обработки
data = get_compared_group(Storage,folder_address,image_format,processing_vec);

% Визуализация результатов сравнения для всех сценариев
build_graphs_group(data);

File: src/group_test.py
import os
import numpy as np
from skimage import io
import matplotlib.pyplot as plt

def get_compared_group(Storage, folder_address, image_format, processing_vec):
    # Placeholder for get_compared_group function
    # This function should compare vector fields from flow files with processing results
    data = []
    for processing in processing_vec:
        # Placeholder for processing logic
        result = processing(Storage)
        data.append(result)
    return data

def build_graphs_group(data):
    # Placeholder for build_graphs_group function
    # This function should visualize the comparison results for all scenarios
    for i, result in enumerate(data):
        plt.figure()
        plt.title(f'Scenario {i+1}')
        plt.imshow(result)  # Placeholder for actual visualization logic
        plt.colorbar()
    plt.show()

# Script for testing a group of processing scenarios on a dataset

# Clear and initialize the data storage class
storage = Storage()

# Define the directory with data and image format
folder_address = '.../demos/group_test'
image_format = '*.tif'

# Define the set of scenarios for calculation
processing_vec = [processing_1, processing_2, processing_3, processing_4, processing_5]

# Compare vector fields from flow files with processing results
data = get_compared_group(storage, folder_address, image_format, processing_vec)

# Visualize the comparison results for all scenarios
build_graphs_group(data)
